Added trigger interaction, and toggle player method

Donahuec opened the pull request:
TriggerInteraction currently can only access the first Interaction component on an object. Not sure how to fix that. 

Also made some modifications to FirstPersonController, very minor just modified the interact text box so that it stretches for longer prompts

Donahuec committed:
Added trigger interaction, and toggle player method

cpimhoff reviewed (changes requested):
`Interaction.TogglePlayer()` seems a poor name and API for this kind of function. For example, what if a object has two interactions which require a player freeze? They'd both call toggle and the player would be free! How about `Interaction.SetPlayerFrozen(bool isFrozen)`?

The more I think about this... Is this a method we want in the `Interaction` base? This is a player behavior, so it probably belongs in `FirstPersonInteractor`. In `Interaction`, we'd then call `this.trigger.SetFrozen(true)` to freeze.

---

Regarding your issue with `TriggerInteraction` not being able to trigger multiple interactions on a GameObject, I would look into swapping around the API, because it is tricky to get the Unity Editor UI to take in specific components into public fields.

Instead of having the user pass an array of `Interaction` objects into `TriggerInteractions. TriggerInteractions`, instead pass a list of game objects to be triggered (`TriggerInteraction.triggeredObjects`).

Then, in `PerformAction`, for each GameObject in `triggeredObjects`, simply use `gameObj.GetComponents<Interaction>( )` to grab all its interactions, then call `Interact()` on them all.

This removes some granularity in selecting _which_ interactions are fired on a game object, but the player always triggers all of them when they press F over an object, so this seems like a reasonable constraint to enforce here.

Donahuec committed:
updated toggleplayer to SetPlayerFrozen, changed TriggerInteraction to accept objects

Donahuec commented:
Alright, I updated TriggerInteraction to take GameObjects instead. I reworked togglePlayer to follow the SetPlayerFrozen idea. I'm hesitant to move it to the FirstPersonInteractor for a couple of reasons. Instead of passing the gameobject to Interact, we would have to pass the firstPersonInteractor object, or something similar to be able to get to a method within it. It also would then cause problems if we wanted to pass a different object as a trigger, which I wanted to leave as an available option.

cpimhoff commented:
We would still pass the generic `GameObject` into `Interact()` and then simply check to see if the game object contains a `FirstPersonInteractor` component, and then use that.

This would allow the structure I'm angling for here, as well as maintain all present flexibility.
